NLog.Windows.Forms/RichTextBoxTarget.cs

```diff
@@ -407,5 +657,41 @@ private void SendTheMessageToRichTextBox(string logMessage, RichTextBoxRowColori
                 textBox.ScrollToCaret();
             }
         }
+
+        /// <summary>
+        /// Stores a new message in internal queue, if it exists. Removes overflowing messages.
+        /// </summary>
+        /// <param name="logMessage">a message to store</param>
+        /// <param name="rule">a corresponding coloring rule</param>
+        private void StoreMessage(string logMessage, RichTextBoxRowColoringRule rule)
+        {
+            lock (messageQueueLock)
+            {
+                if (messageQueue == null)
+                {
+                    return;
+                }
+                if (MaxLines > 0)
+                {
+                    while (messageQueue.Count >= MaxLines)
+                    {
+                        messageQueue.Dequeue();
+                    }
+                }
+                messageQueue.Enqueue(new MessageInfo(logMessage, rule));
+            }
+        }
+
+        private class MessageInfo
+        {
+            internal readonly string message;
```

Comment:
We have hearty discussion on this matter in our company, and I'm giving in to using fields ). Fixed.